packages/date-picker/test/basic.test.js

```diff
@@ -576,3 +592,100 @@ describe('initial value attribute', () => {
     expect(input.value).to.equal('1/1/2000');
   });
 });
+
+describe('auto open disabled', () => {
+  let datepicker, input, toggleButton;
+
+  beforeEach(() => {
+    datepicker = fixtureSync('<vaadin-date-picker value="2000-01-01"></vaadin-date-picker>');
+    input = datepicker.inputElement;
+    toggleButton = datepicker.shadowRoot.querySelector('[part="toggle-button"]');
+    datepicker.autoOpenDisabled = true;
+  });
+
+  it('should focus the input on touch tap', () => {
+    touchTap(input);
+    expect(isFocused(input)).to.be.true;
+  });
+
+  it('should not blur the input on open', async () => {
+    touchTap(input);
+    await open(datepicker);
+    expect(isFocused(input)).to.be.true;
+  });
+
+  it('should blur the input on fullscreen open', async () => {
+    datepicker._fullscreen = true;
+    touchTap(input);
+    await open(datepicker);
+    expect(isFocused(input)).to.be.false;
+  });
+
+  it('should not open on input tap when autoOpenDisabled is true and not on mobile', () => {
+    tap(input);
+    expect(datepicker.opened).not.to.be.true;
+  });
+
+  it('should open by tapping the calendar icon even if autoOpenDisabled is true', async () => {
+    tap(toggleButton);
+    await oneEvent(datepicker.$.overlay, 'vaadin-overlay-open');
+  });
+});
+
+describe('ios', () => {
+  let datepicker, input;
+
+  beforeEach(() => {
+    datepicker = fixtureSync('<vaadin-date-picker value="2000-01-01"></vaadin-date-picker>');
+    input = datepicker.inputElement;
+    datepicker._ios = true;
+  });
+
+  it('should focus the input when closed', () => {
+    datepicker.focus();
+    expect(isFocused(input)).to.be.true;
+  });
+
+  it('should blur the input when opened', async () => {
+    datepicker.focus();
+    await open(datepicker);
+    expect(isFocused(input)).to.be.false;
+  });
+
+  describe('auto open disabled', () => {
+    let toggleButton;
+
+    beforeEach(() => {
+      datepicker.autoOpenDisabled = true;
+      toggleButton = datepicker.shadowRoot.querySelector('[part="toggle-button"]');
+    });
+
+    it('should focus the input on touch tap', () => {
+      touchTap(input);
+      expect(isFocused(input)).to.be.true;
+    });
+
+    it('should blur the input on open', async () => {
+      touchTap(input);
+      await open(datepicker);
+      expect(isFocused(input)).to.be.false;
+    });
+
+    it('should blur the input on fullscreen open', async () => {
+      datepicker._fullscreen = true;
+      touchTap(input);
+      await open(datepicker);
+      expect(isFocused(input)).to.be.false;
+    });
+
+    it('should not open on input tap when autoOpenDisabled is true and not on mobile', () => {
```

Comment:
Not sure about the test purpose. Does the test make sense in the `ios` section? Or should it have a different name?

> and not on mobile

... what does it mean?

Reply:
Yeah, it's the same old test, didn't rename it either. In this test "mobile" means fullscreen mode. You can be on iOS without the date picker being in fullscreen mode (iPad). So the test is about the dropdown not opening when you tap the input in case autoOpenDisabled is on

Reply:
I tried to force the fullscreen mode before tapping the input, but nothing changed in terms of the test success, it still passed.

```js
it('should not open on input tap when ...', () => {
  this._fullscreen = true;
  tap(input);
  expect(datepicker.opened).not.to.be.true;
});
```

Did you mean something different?

Reply:
Ah, is the idea to make sure that the dropdown opens neither in `_fullscreen === true` nor in `_fullscreen === false`? If so. it would probably make sense to have a pair test for `_fullscreen = false`.

Reply:
Done, renamed the test so it makes more sense.

Reply:
Looks clear now, thanks. 👍